### PR TITLE
core(user-flow): i18n default names

### PR DESCRIPTION
**Pull request by adamraine**
Closes https://github.com/GoogleChrome/lighthouse/issues/14049

### CODE REVIEW

File: core/test/user-flow-test.js [hunk 1]
```diff
@@ -32,6 +19,8 @@ await td.replaceEsm('../gather/timespan-runner.js', timespanModule);
 
 const mockRunner = await mockRunnerModule();
 
+const {getStepName, getFlowName, UserFlow, auditGatherSteps} = await import('../user-flow.js');
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply core/test/user-flow-test.js:22]: `// Some imports needs to be done dynamically, so that their dependencies will be mocked.`

**Review by connorjclark — APPROVED** [review]

File: core/test/user-flow-test.js [hunk 2]
```diff
@@ -32,6 +19,11 @@ await td.replaceEsm('../gather/timespan-runner.js', timespanModule);
 
 const mockRunner = await mockRunnerModule();
 
+// Some imports needs to be done dynamically, so that their dependencies will be mocked.
+// See: https://jestjs.io/docs/ecmascript-modules#differences-between-esm-and-commonjs
+//      https://github.com/facebook/jest/issues/10025
```
Comment (brendankenny) [inline core/test/user-flow-test.js:24]: seems weird to still be linking to jest docs for this, though

File: core/user-flow.js [hunk 3]
```diff
@@ -10,9 +10,46 @@ import {startTimespanGather} from './gather/timespan-runner.js';
...
 /** @typedef {WeakMap<LH.UserFlow.GatherStep, LH.Gatherer.FRGatherResult['runnerOptions']>} GatherStepRunnerOptions */
 
+const UIStrings = {
+  /**
+   * @description Default name for a user flow on the given url. "User flow" refers to the series of page navigations and user interactions being tested on the page.
+   * @example {https://example.com} url
+   */
+  defaultFlowName: 'User flow ({url})',
+  /**
+   * @description Default name for a user flow step that analyzes a page navigation.
```
Comment (brendankenny) [inline core/user-flow.js:25]: "user flow" isn't going to mean anything to a translator here, and doesn't really seem to matter for the name anyways. Seems better just to explain what's meant by "navigation" and "report"? 

(same for timespan/snapshot below)

File: core/user-flow.js [hunk 4]
```diff
@@ -10,9 +10,46 @@ import {startTimespanGather} from './gather/timespan-runner.js';
...
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+/**
+ * @param {string} message
+ * @param {Record<string, string | number>} values
+ * @param {LH.Locale} locale
+ */
+function translate(message, values, locale) {
+  const icuMessage = str_(message, values);
+  return getFormatted(icuMessage, locale);
```
Comment (brendankenny) [inline core/user-flow.js:50]: Feels weird to be immediately removing the IcuMessage. Should this be handled by either the flow report's i18n or by the LHR's typical `replaceIcuMessages` flow?

This won't work with swapping locales, but looks like there's nowhere you can swap a flow report's locale anyways.

File: core/test/user-flow-test.js [hunk 5]
```diff
@@ -331,38 +326,38 @@ describe('UserFlow', () => {
         },
       };
 
-      /** @type {LH.UserFlow.GatherStep[]} */
```
Comment (brendankenny) [inline core/test/user-flow-test.js:334]: why not
```
// @ts-expect-error - incomplete mock.
/** @type {LH.UserFlow.GatherStep[]} */
```

or whatever? This file can also just be removed from the tsconfig if mocking continues to make it too annoying to maintain the types.

File: core/test/user-flow-test.js (hunk 2 above)
Comment (adamraine) [reply core/test/user-flow-test.js:24]: It matches everything else so probably good for now

File: core/user-flow.js (hunk 3 above)
Comment (adamraine) [reply core/user-flow.js:25]: I'll do this one in a follow up

File: core/user-flow.js (hunk 4 above)
Comment (adamraine) [reply core/user-flow.js:50]: The messages are designed to be customized, so I want to avoid shipping ICU messages as the step/flow names.

**Pull request merged**